### DISCUSSION TITLE
Should a prompt dialog be shown for MS Windows exe files too?

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2000] [Ideas]
Very sorry for the long comment but you need to read it to understand why I ask the question.

I received an email about the risks of pcmanfm-qt's opening files with their associated apps through command-line. My reply was simple: If opening files with their associated apps is a "risk", what isn't?! Please note that the writer didn't talk about executables, for which pcmanfm-qt shows a prompt dialog. His scenario was like this: a malicious attacker pretends to *Firefox* that a *non-executable* file is a directory and the user sets pcmanfm-qt as the default app for opening directories in Firefox, while pcmanfm-qt sees its real mimetype and gives it to its associated app.

I told him that the associated apps are responsible for processing files and overprotection is a bad idea.

However, the writer's whole point was Wine and `mono` (although he wanted me to see it as a general problem). If I wasn't busy with coding, I'd read the long attachments of the first mail and there would be no need to several replies for me to see that.

Wine may bring the problems of MS Windows to Linux because it executes `application/x-ms-dos-executable`, while the latter is *not* an executable file in Linux.

My reply was (and is) still negative: we shouldn't ruin our command-line because of a special case like that. And my question is, "Should we treat `application/x-ms-dos-executable` as an executable mimetype?"

Please also consider that:

 * If the same prompt is used, people might ask, "How can I remove MS Windows prompts but keep the prompt for Linux executables?" But a new prompt might complicate the code (I haven't looked into it yet).
 * The security risk of Wine can be easily avoided by the user: by associating `application/x-ms-dos-executable` to lxqt-archiver or any archive manager that can handle it. He/she can still use Wine by right clicking exe files.

My opinion is that Wine is Wine and we shouldn't compensate for its issues. I opened this page to know whether I'm in the minority. If I am, I'll do it, although reluctantly.

**Comment by telmobss1** [comment]
I'm not very attentive on the subject but the bottles works, it is increasingly, I do not see why to use Windows programs, since flatpak is increasingly improving as a universal way to install applications in all distros (yes, it is even possible to run Windows programs through wine in flatpak)
